### ISSUE TITLE
Listening for a combined event does not work

### ISSUE THREAD

**Issue by TeemuStenhammar** [erikzenker/inotify-cpp#61]
Hi, 

From this example snippet: 

```
// Set the events to be notified for
  auto events = { Event::open | Event::is_dir, // some events occur in combinations
                  Event::access,
                  Event::create,
                  Event::modify,
                  Event::remove,
                  Event::move };
```

I got the idea that I was able to listen for combination of event. E.g. `Event::open | Event::is_dir` would fire only when a directory was opened. However, this is not the case. An event will be fired with each event that either contains `Event::open` or `Event::is_dir`. This means that the logical or'ing of events is actually redundant now because these two following snippets are actually equal when it comes to listening for events:

```
auto events = { Event::open | Event::is_dir }
```
```
auto events = { Event::open, Event::is_dir }
```

It all boils down to this function in `Event.cpp` :

```
bool containsEvent(const Event& allEvents, const Event& event)
{
    if (static_cast<std::uint32_t>(event & allEvents)) {
        return true;
    } else {
        return false;
    }
}
```

Which basically just checks that any bit is up.

So, I would have liked to listen for events of directories being created and this is not possible with the current API without listening for all events and filtering them manually. Example code I presented in the beginning made me think listening for combination events is possible, but that was not the case.

What was the intended use of the or'ing of events? Is this a bug or wanted behaviour?

**Comment by erikzenker** [comment]
Hi @TeemuStenhammar, thx for using the library. Sounds like a Bug. It should be possible to listen for the described combonation of events. I am currently on xmas holidays and will have a detailed look in a couple of days.

**Comment by erikzenker** [comment]
@TeemuStenhammar I created a pull request with a fix and a proper test (see #62)
